website/src/auth.tsx

```diff
@@ -64,6 +68,54 @@ export const UserProvider = (props: { children: any }) => {
     return
   }, [user])
 
+  function createUser(username: string, password: string) {
+    console.log(`requesting adding user ${username} to Cognito User Pool`)
+    let userAttributes = [{ Name: "email", Value: username }].map((attr) => {
+      return new CognitoUserAttribute(attr)
+    })
+    userPool.signUp(
+      username,
+      password,
+      userAttributes,
+      [],
+      async (err, result) => {
+        if (err) {
+          alert(err.message || JSON.stringify(err))
+          return
+        }
+        let cognitoUser = result?.user
+        if (!cognitoUser) {
+          console.log("failed to create user and activate session")
+          return
+        }
+        setUser(cognitoUser)
+        await navigate("/auth/confirm-signup")
+      }
+    )
+  }
+
+  function resetVerificationCode() {
+    user?.resendConfirmationCode((err, result) => {
+      if (err) {
+        alert(err.message || JSON.stringify(err))
+        return
+      }
+      console.log(result)
+    })
+  }
+
+  function verifyUser(verificationCode: string) {
+    user?.confirmRegistration(verificationCode, false, (err, result) => {
+      if (err) {
+        alert(err.message || JSON.stringify(err))
+        return
+      }
+      console.log(result)
+      setUser(user)
+      navigate("/auth/login") 
+    })
+  }
```

Comment:
Something here feels weird to me but if this works I'm sure it's for a reason. There seem to be some weird things about verified users we may want to shelter the rest of the frontend from. Based on this code it seems I can be "logged in" and `useUser` will get me the user I'm logged in as **even if I still haven't verified my email** - that feels ungood.

Also this `setUser(user)` sort of scares me. Does this flow work without this? Why not?